src/main/java/com/mraof/minestuck/inventory/captchalogue/CaptchaDeckContainer.java

```diff
@@ -34,12 +36,8 @@ public void setContainerItem(ItemStack stack)
 		inventory.setItem(0, stack);
 	}
 	
-	private void addSlots(Inventory playerInventory)
+	private void addSlots()
```

Comment:
It doesn't seem accurate to move the player inventory slots from this function directly into the constructor.
I'd suggest moving it back by calling `ContainerHelper.addPlayerInventorySlots()` in `addSlots()`, or alternatively to inline the `addSlots()` helper function.

Reply:
I don't remember touching the captchalogue class

Reply:
oh right the helper function !

Reply:
```java
private void addSlots(Inventory playerInventory)
	{
		ContainerHelper.addPlayerInventorySlots(this::addSlot, 9, 63, playerInventory);
		addSlot(new Slot(this.inventory, 0, 81, 32)
		{
			@Override
			public boolean mayPlace(ItemStack stack)
			{
				return ModusTypes.getTypeFromItem(stack.getItem()) != null || stack.getItem().equals(MSItems.CAPTCHA_CARD) && !AlchemyHelper.isPunchedCard(stack);
			}
		});
	}
	```
	Would this work for you?

Reply:
Yep, that's what I mean by
> moving it back by calling `ContainerHelper.addPlayerInventorySlots()` in `addSlots()`